File: csharp/src/Ice/Internal/Acceptor.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.Internal;
public interface Acceptor
{
    void close();
    EndpointI listen();
    bool startAccept(AsyncCallback callback, object state);
    void finishAccept();
    Transceiver accept();
    string protocol();
    string ToString();
    string toDetailedString();
}

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/Acceptor.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

import java.nio.channels.ServerSocketChannel;

/**
 * @hidden Public because it's used by IceBT and SSL.
 */
public interface Acceptor {
    ServerSocketChannel fd();

    void setReadyCallback(ReadyCallback callback);

    void close();

    EndpointI listen();

    Transceiver accept();

    String protocol();

    @Override
    String toString();

    String toDetailedString();
}
